test/ctia/stores/es/mapping_test.clj

```diff
@@ -4,229 +4,185 @@
-                                              {:range {:timestamp
-                                                       {"gte" "2019-07-15T01:00:00.000-00:00"}}}
-                                              nil
-                                              {})]
-        (test-sort "timestamp" '("doc0" "doc1" "doc2"))
-        (is (nil? (-> search-res-all :data seq)))
-        (is (= #{"doc1" "doc2"}
-               (->> search-res-range
-                    :data
-                    (map :id)
-                    set)))))
-    (testing "float-type should not be included in _all"
-      (test-sort "float" '("doc0" "doc1" "doc2"))
-      (test-not-all "float" 100 "doc0"))
-    (testing "long-type should not be included in _all and we can be sorted on"
-      (test-sort "long" '("doc0" "doc1" "doc2"))
-      (test-not-all "long" 200 "doc0"))
-    (testing "integer-type should not be included in _all"
-      (test-sort "integer" '("doc0" "doc1" "doc2"))
-      (test-not-all "integer" 300 "doc0"))
-    (testing "boolean-type should not be included in _all"
-      (test-sort "boolean,id" '("doc1" "doc0" "doc2"))
-      (test-not-all "boolean" false "doc1"))
-    (testing "embedded data table should not be included in _all"
-      (test-not-all "table.row_count" 0 "doc0"))
-    (index/delete! es-conn indexname)))
+                                      nil)
+             ]
+         (is (= 3
+                (count res-text-1)
+                (count res-text-2)
+                (count res-cross-field-text)))
+         (is (= res-text-1
+                res-text-2
+                res-cross-field-text))
+         (is (nil? (seq res-missing-text)))
+
+         (is (thrown? clojure.lang.ExceptionInfo
+                      (search {:query_string {:query "simpletext"}}
+                              nil
+                              {:sort_by "sortable-text"
+                               :sort_order "asc"})))
+         (test-sort "sortable-text.whole" '("doc0" "doc1" "doc2"))))
+     (testing "ts mapping should be a date type, not in _all field and sortable"
+       (let [res-all (search {:query_string {:query "2019"}}
+                             nil
+                             {})
+             res-range (search {:range {:timestamp
+                                        {"gte" "2019-07-15T01:00:00.000-00:00"}}}
+                               nil
+                               {})]
+         (test-sort "timestamp" '("doc0" "doc1" "doc2"))
+         (is (nil? (seq res-all)))
+         (is (= #{"doc1" "doc2"}
+                (->> res-range
+                     (map :id)
+                     set)))))
+     (testing "scalar type should be properly sorted"
+       (test-sort "float" '("doc0" "doc1" "doc2"))
+       (test-sort "long" '("doc0" "doc1" "doc2"))
+       (test-sort "integer" '("doc0" "doc1" "doc2"))
+       (test-sort "boolean,id" '("doc1" "doc0" "doc2"))
+       (test-sort "table.row_count" '("doc0" "doc1" "doc2"))))))
```

Comment:
Looks like a good candidate for `are`

Reply:
I agree, but indeed test-sort performs 2 `is` testing ascending and descending order option for `search-docs`(it is also used above on other token and text mapping type).

Reply:
Very minor opinion, no action asked: since `test-sort` is a function, `are` won't improve error messages. I might use `doseq` to save the redundant data and open the door to generate more cases in the future.

```clojure
(doseq [[s r] (concat
                (map vector
                     ["float"
                      "long"
                      "integer"
                      "table.row_count"]
                     (repeat '("doc0" "doc1" "doc2")))
                ;; could generate combinations of cases here
                ...
                ;; could hardcode more cases here
                ...)]
  (test-sort s r))
```